EosPayload/lib/config.py

```diff
@@ -0,0 +1,186 @@
+import inspect
+import json
+import logging
+import os
+import re
+
+from EosLib import device
+
+from EosPayload import drivers
+from EosPayload.lib.base_drivers.driver_base import DriverBase
+from EosPayload.lib.orcheostrator.device_container import DeviceContainer, Status
+
+base_drivers = ["DriverBase", "PositionAwareDriverBase"]
+
+
+class OrcheostratorConfig:
+    def __init__(self):
+        self.global_config = {}
+        self.enabled_devices = []
+
+class OrcheostratorConfigParser:
+
+    config_indent = "\t"
+
+    def __init__(self, logger: logging.Logger, config_filepath: str):
+        self.logger = logger
+        self.config_filepath = config_filepath
+        self.orcheostrator_config = OrcheostratorConfig()
+        self.used_device_ids = []
+        self.valid_driver_classes = {}
+        self.disabled_drivers = []
+        self.unused_drivers = {}
+
+    def get_raw_config(self) -> dict | None:
+        try:
+            with open(self.config_filepath) as config_file:
+                raw_config = json.load(config_file)
+        except OSError:
+            self.logger.critical("Unable to open config file")
+            return None
+        self.logger.info(f"Opened config file at {os.path.abspath(self.config_filepath)}")
+        return raw_config
+
+    @staticmethod
+    def valid_driver(driver) -> bool:
+        """ Determines if given class is a valid driver.
+
+        :param driver: the class in question
+        :return: True if valid, otherwise False
+        """
+        return (
+                (driver is not None)
+                and inspect.isclass(driver)
+                and inspect.getmodule(driver).__name__.startswith("EosPayload.drivers")
+                and issubclass(driver, DriverBase)
+        )
+
+    @staticmethod
+    def collect_valid_driver_classes() -> dict[str, DriverBase]:
+        valid_driver_classes = {}
+        for attribute_name in dir(drivers):
+            driver = getattr(drivers, attribute_name)
+            if OrcheostratorConfigParser.valid_driver(driver):
+                valid_driver_classes.update({attribute_name: driver})
```

Comment:
Because of how dir works, this would log every variable in drivers, which would kinda clog up the logs. I'm currently considering revamping how finding drivers works though so it may soon be viable.